release/e2e.py

```diff
@@ -792,6 +792,31 @@ def report_result(
     )
     logger.info("Result has been persisted to the database")
 
+    # TODO(jjyao) Migrate to new infra later
+    logger.info("Persisting results to the databricks delta lake...")
+
+    result_json = {
+        "_table": "release_test_result",
+        "created_on": now.strftime("%Y-%m-%d %H:%M:%S"),
+        "status": status,
+        "results": results,
+        "test_name": test_name,
+        "team": team,
+        "cluster_url": results["_session_url"],
+        "wheel_url": results["_commit_url"],
+        "runtime": results["_runtime"],
+        "stable": results["_stable"],
+    }
+
+    logger.debug(f"Result json: {json.dumps(result_json)}")
+
+    firehose_client = boto3.client("firehose", region_name="us-west-2")
+    firehose_client.put_record(
```

Comment:
Do we need exponential retry here? (like S3 APIs we use)

Reply:
+1, or at least catch errors here